addons/stock/report/report_stockpicking_operations.xml

```diff
@@ -66,100 +50,141 @@
-                                        </td>
-                                        <td class=" text-center h6" t-if="has_serial_number">
-                                            <span t-if="has_serial_number and (ml.lot_id or ml.lot_name)" t-esc="ml.lot_id.name or ml.lot_name" t-options="{'widget': 'barcode', 'humanreadable': 1, 'width': 400, 'height': 100, 'img_style': 'width:100%;height:35px;'}">
-                                                <div class="bg-light border-1 rounded d-flex flex-column align-items-center justify-content-center px-1 py-2 opacity-75 text-muted text-center">
-                                                    (serial barcode)
+                                            </td>
+                                            <td t-if="from_col_exists" groups="stock.group_stock_multi_locations">
+                                                <div t-if="not move_has_serial_number">
+                                                    <span t-field="move.location_id.display_name">WH/Stock</span>
+                                                    <t t-if="move.move_line_ids and move.move_line_ids[0].package_id">
+                                                        <span t-field="move.move_line_ids[0].package_id">Package A</span>
+                                                    </t>
                                                 </div>
-                                            </span>
-                                        </td>
-                                        <td class="text-center" t-if="has_barcode">
-                                            <t t-if="product_barcode != ml.product_id.barcode">
-                                                <span t-if="ml.product_id and ml.product_id.barcode">
-                                                    <div t-field="ml.product_id.barcode" t-options="{'widget': 'barcode', 'symbology': 'auto', 'width': 400, 'height': 100, 'quiet': 0, 'img_style': 'height:35px;'}">
+                                            </td>
+                                            <td t-if="to_col_exists" groups="stock.group_stock_multi_locations">
+                                                <div t-if="not move_has_serial_number">
+                                                    <span t-field="move.location_dest_id.display_name">WH/Outgoing</span>
+                                                    <t t-if="move.move_line_ids and move.move_line_ids[0].result_package_id">
+                                                        <span t-field="move.move_line_ids[0].result_package_id">Shelf 1</span>
+                                                    </t>
+                                                </div>
+                                            </td>
+                                            <td class="text-center" t-if="picking_has_barcode or picking_has_serial_number" style="background-color: #FFF;">
```

Comment:
I checked that the last column is grey only in [`boxed_layout`](https://github.com/odoo/odoo/blob/24c79ab2d6a8929ffdd3e48fbdeef5c6901c704e/addons/web/static/src/webclient/actions/reports/layout_assets/layout_boxed.scss#L77). Apparently, it can be disabled by adding `o_main_table` class to the table.